lib/types.nix

```diff
@@ -0,0 +1,114 @@
+{ base16 }:
+{ config, lib, options, ... }@args:
+with lib;
+with config.lib.stylix;
+{
+  config.lib.stylix = {
+    static = mkOptionType {
+      name = "static";
+      description = "static wallpaper";
+      descriptionClass = "noun";
+      check = isStatic;
+    };
+    isStatic = value:
+      builtins.attrNames value == [ "colors" "image" ];
+
+    animation = mkOptionType {
+      name = "animation";
+      description = "animated wallpaper";
+      descriptionClass = "noun";
+      check = isAnimation;
+    };
+    isAnimation = value:
+      builtins.attrNames value == [ "animation" "colors" "image" ];
+
+    video = mkOptionType {
+      name = "video";
+      description = "video wallpaper";
+      descriptionClass = "noun";
+      check = isVideo;
+    };
+    isVideo = value:
+      builtins.attrNames value == [ "colors" "image" "video" ];
+
+    slideshow = mkOptionType {
+      name = "slideshow";
+      description = "slideshow wallpaper";
+      descriptionClass = "noun";
+      check = isSlideshow;
+    };
+    isSlideshow = value:
+      builtins.attrNames value == [ "colors" "delay" "image" "images" ];
+
+    scheme = mkOptionType {
```

Comment:
This replaces the old `config.lib.stylix.scheme` which was effectively an alias to `config.lib.stylix.colors`. For backwards compatibility we could move all the types into `config.lib.stylix.types`, then it will be possible to bring back `scheme` as an alias to the new `config.stylix.colors` option, with a warning similar to what `config.lib.stylix.colors` does now.

Reply:
Fixed in 5d666dc885f430566f7204fcba1c5f39a9b97ac6